lte/gateway/c/core/oai/test/n11/test_smf_service_client.cpp

```diff
@@ -75,8 +75,7 @@ TEST(test_create_sm_pdu_session_v4, create_sm_pdu_session_v4) {
       rat_req->mutable_gnode_endpoint()->end_ipv4_addr());
   uint8_t* pti_decoded = (uint8_t*) rat_req->procedure_trans_identity().c_str();
   EXPECT_TRUE(pti == *pti_decoded);
-  EXPECT_TRUE(
-      ipv4_addr == rat_req->mutable_pdu_address()->redirect_server_address());
+  EXPECT_TRUE(ipv4_addr == req_cmn->ue_ipv4());
```

Comment:
please go through all [my comments](https://github.com/magma/magma/pull/10318/files#r749786178). You fixed one of my concerns, but did not take care of using different IPs for each field. 

As described before if you use the same IP you may be hiding bugs (for example mixing IPs which is what this PR is fixing)

Reply:
Why is this test checking  redirect ip at all. The function call arguments do not include any redirect field 

I also see on the function we are setting the redirect address to the same subscriber IP. What is this for?!

```
  // Set the PDU Address
  req_rat_specific->mutable_pdu_address()->set_redirect_server_address(
      (char*) ipv4_addr);
```

https://sourcegraph.com/-/editor?remote_url=git%40github.com%3Auri200%2Fmagma.git&branch=federated_lte_test_add_client_checkin&file=lte%2Fgateway%2Fc%2Fcore%2Foai%2Flib%2Fn11%2FSmfServiceClient.cpp&editor=JetBrains&version=v1.2.1&utm_product_name=CLion&utm_product_version=2021.1.2&start_row=101&start_col=22&end_row=101&end_col=22

Reply:
We had used `redirect_server_address` for UE IPv4 address , now, we found this is setting in wrong message for UE IPv4 address and this PR is a fix for it.

Reply:
Yes @mehul-jindal-wavelabs  that part is understood. 

The question is, why does the create request has a `redirect_server_address` at all?

Reply:
@uri200 
1. Base code reused RedirectServer structure to hold UE PDU address, 
   sessiond used the same parameter to set up IPv4 PDU Session.
   
   session_manager.proto
   ```
    // PDU session Establishment from AMF to SMF
	// SET MESSSAGE
	message M5GSMSessionContext {
	   ---
	    // IP address. Imported from session_manager.proto file
		RedirectServer pdu_address = 3;
	   ---
	}
   ```
2. we are enhancing code to support IPv6 PDU Sessions.
   CommonSessionContext has a placeholder for both ipv4 and ipv6.
   PDU Session IP place holder moved to CommonSessionContext.
   we have done code change accordingly. 
   now M5GSMSessionContext { RedirectServer pdu_address = 3; } unused.

	```
	message SetSMSessionContext {
	  // Common message for 4g, 5g and WiFi.
	  CommonSessionContext common_context = 1;
	  RatSpecificContext   rat_specific_context = 2;
	}

	message CommonSessionContext {
	  string ue_ipv4 = 3;
	  string ue_ipv6 = 10;
	  ....	
	}

	```
3. we have done cleanup below parameter-related code in amf & smf.
   M5GSMSessionContext { RedirectServer pdu_address = 3; }

Reply:
Hi @uri200,  
1. Base code reused RedirectServer structure to hold UE PDU address, 
   sessiond used the same parameter to set up IPv4 PDU Session.
   
   session_manager.proto
   ```
    // PDU session Establishment from AMF to SMF
	// SET MESSSAGE
	message M5GSMSessionContext {
	   ---
	    // IP address. Imported from session_manager.proto file
		RedirectServer pdu_address = 3;
	   ---
	}
   ```
2. we are enhancing code to support IPv6 PDU Sessions.
   CommonSessionContext has a placeholder for both ipv4 and ipv6.
   PDU Session IP place holder moved to CommonSessionContext.
   we have done code change accordingly. 
   now M5GSMSessionContext { RedirectServer pdu_address = 3; } unused.

	```
	message SetSMSessionContext {
	  // Common message for 4g, 5g and WiFi.
	  CommonSessionContext common_context = 1;
	  RatSpecificContext   rat_specific_context = 2;
	}

	message CommonSessionContext {
	  string ue_ipv4 = 3;
	  string ue_ipv6 = 10;
	  ....	
	}

	```
3. we have done cleanup below parameter-related code in amf & smf.
   M5GSMSessionContext { RedirectServer pdu_address = 3; }

Reply:
@uri200 , we have done cleanup redirect_server_address  related code.